gcp/table_gcp_storage_object.go

```diff
@@ -0,0 +1,349 @@
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "time_storage_class_updated",
+				Description: "The time at which the object's storage class was last changed.",
+				Transform:   transform.FromGo().NullIfZero(),
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "updated",
+				Description: "The modification time of the object metadata.",
+				Transform:   transform.FromGo().NullIfZero(),
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+
+			// JSON fields
+			{
+				Name:        "acl",
+				Description: "Access controls on the object.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "customer_encryption",
+				Description: "Metadata of customer-supplied encryption key, if the object is encrypted by such a key",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "owner",
+				Description: "The owner of the object. This will always be the uploader of the object.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "iam_policy",
+				Description: "An Identity and Access Management (IAM) policy, which specifies access controls for Google Cloud resources. A `Policy` is a collection of `bindings`. A `binding` binds one or more `members` to a single `role`. Members can be user accounts, service accounts, Google groups, and domains (such as G Suite). A `role` is a named list of permissions; each `role` can be an IAM predefined role or a user-created custom role. For some types of Google Cloud resources, a `binding` can also specify a `condition`, which is a logical expression that allows access to a resource only if the expression evaluates to `true`.",
+				Hydrate:     getStorageObjectIAMPolicy,
+				Transform:   transform.FromValue(),
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// standard steampipe columns
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getObjectAka,
+				Transform:   transform.FromValue(),
+			},
+
+			// standard GCP columns
+			{
+				Name:        "project",
+				Description: ColumnDescriptionProject,
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     plugin.HydrateFunc(getProject).WithCache(),
+				Transform:   transform.FromValue(),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listStorageObjects(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	bucket := d.EqualsQualString("bucket")
+
+	// The bucket name should not be empty
+	if bucket == "" {
+		return nil, nil
+	}
+
+	service, err := StorageService(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Trace("gcp_storage_object.listStorageObjects", "connection_error", err)
+		return nil, err
+	}
+
+	projection := "noAcl"
```

Comment:
Is there a reason not to always set `project = "full"`? Is there other info we don't get if `project = "noAcl"`? Also, are there other values `project` can be?

Reply:
The values are only `noAcl` and `full`.

It does make sense, @cbruno10, to set it always as `full`.
I tried to keep it as per `gcp_storage_bucket`, where we choose `full` only when `acl` is a selected column.

Reply:
In the `gcp_storage_bucket`, should we change the behaviour there? Any performance impacts from always setting it to `full`?

Reply:
Observation with `full` in `gcp_storage_bucket`
Time: 3.0s. Rows fetched: 17. Hydrate calls: 0.

Observation without `full` in `gcp_storage_bucket `:
Time: 2.6s. Rows fetched: 17. Hydrate calls: 0.

Not very significant for less number of buckets.

It might make a difference in `gcp_storage_object` if the count is more bigger(which I think would be expected).

Reply:
With caching enabled, if you run a query without `acl` in the select statement, and then a query with it included, do you get the correct results? I'm OK to leave the check in, as long as caching works as intended.

Reply:
Hmm, had caching disabled while testing.
It doesn't work as expected with caching on.
Have updated the `projection` to remain `full` always for both tables.

Reply:
We ran into a similar issue when we tried omitting JSON columns from GraphQL queries, so this felt like a possibility here too.